### PR TITLE
[FEAT] Move shortener list to come from a plain file, Closes #14

### PR DESCRIPTION
**Pull request by AvnerCohen**
@mudge Following up on the discussion at #14 , This PR adds the logic to load the list of shorteners from a plain file, based, nearly one to one, on your suggestions.

There are 2 additional changes:

1. I've added .bundle to the ignore file (created if you have a special configuration for the bundler cli, as in my case, and seems like a reasonble default).
2. Removed the custom any? implementation of ```include?``` to use the built in set.include? rather than running the regular expression on for each domain.
This makes the test suite run X2:
```sh
› for i in {1..10}; do bundle exec rspec | grep Finished ; done
Finished in 0.11635 seconds (files took 0.1828 seconds to load)
Finished in 0.11136 seconds (files took 0.20086 seconds to load)
Finished in 0.11364 seconds (files took 0.20566 seconds to load)
Finished in 0.11202 seconds (files took 0.18655 seconds to load)
Finished in 0.11531 seconds (files took 0.19866 seconds to load)
Finished in 0.11331 seconds (files took 0.19342 seconds to load)
Finished in 0.14938 seconds (files took 0.25272 seconds to load)
Finished in 0.12448 seconds (files took 0.19337 seconds to load)
Finished in 0.12069 seconds (files took 0.2027 seconds to load)
Finished in 0.1156 seconds (files took 0.19347 seconds to load)
› git checkout shorteners_from_file
Switched to branch 'shorteners_from_file'
› for i in {1..10}; do bundle exec rspec | grep Finished ; done
Finished in 0.05276 seconds (files took 0.19224 seconds to load)
Finished in 0.05239 seconds (files took 0.20213 seconds to load)
Finished in 0.05505 seconds (files took 0.18758 seconds to load)
Finished in 0.0529 seconds (files took 0.19548 seconds to load)
Finished in 0.05358 seconds (files took 0.18276 seconds to load)
Finished in 0.05798 seconds (files took 0.19329 seconds to load)
Finished in 0.05503 seconds (files took 0.20827 seconds to load)
Finished in 0.05771 seconds (files took 0.20531 seconds to load)
Finished in 0.05464 seconds (files took 0.20599 seconds to load)
Finished in 0.05595 seconds (files took 0.20314 seconds to load)
```

I understand there is something obvious I might be missing in terms of why this was added to begin with but considering we just want to match domain names + all test suite is passing, I am not seeing this scenario yet.

### CODE REVIEW

File: lib/embiggen/configuration.rb [hunk 1]
```diff
@@ -18,52 +18,12 @@ def self.redirects
...
-        yatuc.com ye.pe yep.it yfrog.com yhoo.it yiyd.com youtu.be yuarel.com
-        z0p.de zi.ma zi.mu zipmyurl.com zud.me zurl.ws zz.gd zzang.kr ›.ws
-        ✩.ws ✿.ws ❥.ws ➔.ws ➞.ws ➡.ws ➨.ws ➯.ws ➹.ws ➽.ws
-      ))
+      @shorteners ||= ShortenerList.new(shorteners_from_file)
+    end
+
+    def self.shorteners_from_file
+      file_path = File.expand_path('../../../shorteners.txt', __FILE__)
+      File.readlines(file_path).map(&:chomp!)
```
Comment (mudge) [inline lib/embiggen/configuration.rb]: If `chomp!` does not remove a newline, it returns `nil` (e.g. if the file has no trailing newline):

```ruby
[1] pry(main)> File.readlines('foo.txt')
=> ["foo\n", "bar"]
[2] pry(main)> File.readlines('foo.txt').map(&:chomp!)
=> ["foo", nil]
```

File: lib/embiggen/shortener_list.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ def initialize(domains)
     end
 
     def include?(uri)
-      domains.any? { |domain| uri.host =~ /\b#{domain}\z/i }
+      domains.include?(uri.host)
```
Comment (mudge) [inline lib/embiggen/shortener_list.rb]: The regular expression ensures that domains in the list are anchored to the end of the host and are separated by a word boundary, without it `foobar.com` would be matched if `bar.com` is in the list.

**Comment by AvnerCohen** [comment]
Thanks @mudge for the clarification.

1. I have reverted the change for now, I think it's worth exploring, I'll do so in a separate PR though.
That being said, without getting into the details of why (I don't know yet..), I would expect the tests to fail on my PR and they didn't. I've now added test that does fail on such edge cases

2. chomp! vs chomp is fixed, missed that :(

(I'll squash once we are done here)

File: spec/embiggen/uri_spec.rb [hunk 3]
```diff
@@ -251,7 +251,24 @@ module Embiggen
...
+        expect(uri).to_not be_shortened
+      end
+
+      it 'returns false if the link is not shortened but uses a similar ' \
+         'domain without protocol' do
+        uri = described_class.new('notbit.ly/1ciyUPh')
+        expect(uri).to_not be_shortened
+      end
+
+      it 'returns false if the link is not shortened but uses a same TLD ' \
```
Comment (mudge) [inline spec/embiggen/uri_spec.rb]: As TLD stands for "top-level domain" and refers to the `.com` or `.co`, would it be more accurate to change this to `returns false if the link is not shortened but uses the same domain with a different TLD`?

**Comment by mudge** [comment]
Small nitpick on the test description but otherwise this looks good.

**Comment by AvnerCohen** [comment]
squashed and updated - spec title updated as per suggestion.
